scripts/icosahedron_tables.py

```diff
@@ -0,0 +1,173 @@
+#
+# Now multiply by (2/5 * sqrt(5)), using auto-simplify
+# let m = (2/5 * sqrt(5))
+# cxA = m * c1 = 1/2 - sqrt(5)/10
+# cxB = m * c2 = 1/2 + sqrt(5)/10
+# syA = m * s1 = 1/10 * sqrt( 10 * (5 + sqrt(5)) )
+# syN = m * s2 = 1/10 * sqrt( 10 * (5 - sqrt(5)) )
+
+# Height of pentagon within the icosahedron
+hei = 1.0/sqrt(5.0)
+
+# Circumcircle radius of pentagon
+cmr = 2.0/5.0 * sqrt(5.0)
+
+# X and Y coordinates used to construct pentagons
+cxa = 1.0/2.0 - sqrt(5.0)/10.0
+cxb = 1.0/2.0 + sqrt(5)/10.0
+sya = 1.0/10.0 * sqrt( 10.0 * (5.0+sqrt(5.0)) )
+syb = 1.0/10.0 * sqrt( 10.0 * (5.0-sqrt(5.0)) )
+
+icosahedron_vrtx = [
+    ( 0.0,    0.0,    1.0), # 0 top point
+    ( cmr,    0.0,    hei), # 1 top pentagon
+    ( cxa,   -sya,    hei), # 2
+    (-cxb,   -syb,    hei), # 3
+    (-cxb,    syb,    hei), # 4
+    ( cxa,    sya,    hei), # 5
+    (-cmr,    0.0,   -hei), # 6 bottom pentagon
+    (-cxa,   -sya,   -hei), # 7
+    ( cxb,   -syb,   -hei), # 8
+    ( cxb,    syb,   -hei), # 9
+    (-cxa,    sya,   -hei), # 10
+    ( 0.0,    0.0,   -1.0)  # 11 bottom point
+];
+
+print("inline constexpr std::array<osp::Vector3d, 12> gc_icoVrtxPos\n"
+       + "{{\n"
+       + ",\n".join("    {"+nstr_double(x)+", "+nstr_double(y)+", "+nstr_double(z)+"}" for (x, y, z) in icosahedron_vrtx) + "\n"
+       + "}};")
+
+# -----------------------------------------------------------------------------
+
+# Edge lengths vs subdiv level
+
+edge_length_min = [2.0*sin(0.5*acot(0.5) * 0.5**x) for x in range(0, 10)]
+
+print("inline constexpr std::array<float, 10> const gc_icoMinEdgeVsSubdiv\n"
+      + "{\n"
+      + "    " + ", ".join(nstr_float(edge_length) for edge_length in edge_length_min) + "\n"
+      + "};\n")
+
+
+edge_length_max = []
+
+# Any initial triangle from the icosahedron works here
+tri = (
+    ( 0.0,    0.0,    1.0),
+    ( cmr,    0.0,    hei),
+    ( cxa,   -sya,    hei)
+)
+
+for _ in range(0, 10):
+
+    edge_length_max.append(distance(tri[0], tri[1]))
+
+    # Middle triangle is the largest, so it's the maximum
+    tri = (
+        norm(mid(tri[0], tri[1])),
+        norm(mid(tri[1], tri[2])),
+        norm(mid(tri[2], tri[0]))
+    )
+
+print("inline constexpr std::array<float, 10> const gc_icoMaxEdgeVsSubdiv\n"
+      + "{\n"
+      + "    " + ", ".join(nstr_float(edge_length) for edge_length in edge_length_max) + "\n"
+      + "};\n")
+
+# -----------------------------------------------------------------------------
+
+# Tower height
+# If identical towers were built on the two vertices spanning an edge, this is how high each tower
+# needs to be in order to see each other over the horizon.
+#  = exsec(0.5*acot(0.5) * levels)
+tower_heights = [sec(0.5*acot(0.5) * 0.5**x ) - 1 for x in range(0, 10)]
+
+print("inline constexpr std::array<float, 10> const gc_icoTowerOverHorizonVsSubdiv\n"
+      + "{\n"
+      + "    " + ", ".join(nstr_float(edge_length) for edge_length in tower_heights) + "\n"
+      + "};\n")
```

Comment:
Noting the existence of this python script. The icosahedron stuff here use to be calculated at runtime, but only with doubles and implementation-defined std::sqrt. Precalculating all this with much much higher precision is a far more practical